MekHQ/src/mekhq/campaign/mission/enums/ScenarioType.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq.campaign.mission.enums;
+
+import megamek.logging.MMLogger;
+
+public enum ScenarioType {
+    NONE,
+    SPECIAL_LOSTECH,
+    SPECIAL_RESUPPLY;
+
+    /**
+     * @return {@code true} if the scenario is considered a LosTech scenario, {@code false} otherwise.
+     */
+    public boolean isLosTech() {
+        return this == SPECIAL_LOSTECH;
+    }
+
+    /**
+     * @return {@code true} if the scenario is considered a Resupply scenario, {@code false} otherwise.
+     */
+    public boolean isResupply() {
+        return this == SPECIAL_RESUPPLY;
+    }
+
+    public static ScenarioType fromOrdinal(int ordinal) {
+        for (ScenarioType scenarioType : values()) {
+            if (scenarioType.ordinal() == ordinal) {
+                return scenarioType;
+            }
+        }
+
+        final MMLogger logger = MMLogger.create(ScenarioType.class);
+        logger.error(String.format("Unknown Scenario Type ordinal: %s - returning NONE.", ordinal));
+
+        return NONE;
+    }
```

Comment:
make it throw an InvalidParameter or something like that instead, and have the function that is handling it implement an error handling that will force the NONE instead.
This way you avoid adding unnecessary things inside the enum (MMLogger) and you know exactly what it will do.

Or keep the NONE return as default but keep the knowledge of what it means to receive a NONE for scenario type in the function that is handling it. 
I prefer the exception tough, because this means the person HAS to implement something, also allows us to not have the NONE enum value and the user will have to either accept a null value in whatever they are implementing or they will have to add some new state in the enum.

Reply:
I'm not sure I follow. Would you be able to expand on this point? :)

Reply:
```java
    
    int ordinal = 12;
    ScenarioType ScenarioType = null;
    try {
        scenarioType = ScenarioType.values()[ordinal]
    }  catch (IndexOutOfBoundsException e) {
        scenarioType = ScenarioType.FOOBAR
        logger.warn("Unknown scenario type {}, valid scenario types are", ordinal, ScenarioType.values())
    }        
```

or you can do

```java
        // returns a default value if the ordinal is wrong.
        public static ScenarioType fromOrdinal(int ordinal, ScenarioType defaultValue) {
        for (ScenarioType scenarioType : values()) {
            if (scenarioType.ordinal() == ordinal) {
                return scenarioType;
            }
        }
        return defaultValue;
    }      
```

Reply:
I prefer the first, because the function that needs that value should know what are the possible values it can have or handle, so it shuold also be the place where errors are dealt with, this is important mainly when we are dealing with backward compatibility, like trying to load a new type of scenario in an old version of the code, or an old file in a new code. The loader process has to be as lenient as possible with what it reads, and it should handle its own mistakes and errors.